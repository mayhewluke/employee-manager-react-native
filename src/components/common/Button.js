import React from "react";
import { TouchableOpacity, Text } from "react-native";
import PropTypes from "prop-types";

const styles = {
  buttonStyle: {
    flex: 1,
    alignSelf: "stretch",
    backgroundColor: "#fff",
    borderRadius: 5,
    borderWidth: 1,
    borderColor: "#007aff",
    marginLeft: 5,
    marginRight: 5
  },
  textStyle: {
    alignSelf: "center",
    color: "#007aff",
    fontSize: 16,
    fontWeight: "600",
    paddingTop: 10,
    paddingBottom: 10
  }
};

const Button = ({ onPress, children }) => {
  const { buttonStyle, textStyle } = styles;
  return (
    <TouchableOpacity style={buttonStyle} onPress={onPress}>
      <Text style={textStyle}>{children}</Text>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export { Button };
