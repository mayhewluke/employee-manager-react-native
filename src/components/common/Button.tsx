import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

const styles = StyleSheet.create({
  buttonStyle: {
    backgroundColor: "#fff",
    borderColor: "#007aff",
    borderRadius: 5,
    borderWidth: 1,
    marginLeft: 5,
    marginRight: 5,
  },
  textStyle: {
    alignSelf: "center",
    color: "#007aff",
    fontSize: 16,
    fontWeight: "600",
    paddingBottom: 10,
    paddingTop: 10,
  },
});

const Button: React.SFC<{ onPress: () => void }> = ({ onPress, children }) => {
  const { buttonStyle, textStyle } = styles;
  return (
    <TouchableOpacity style={buttonStyle} onPress={onPress}>
      <Text style={textStyle}>{children}</Text>
    </TouchableOpacity>
  );
};

export default Button;
