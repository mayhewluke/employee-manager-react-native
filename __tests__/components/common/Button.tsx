import "jest-enzyme";
import React from "react";

import { shallow } from "enzyme";
import reactTestRenderer from "react-test-renderer";

import Button, { IProps } from "../../../src/components/common/Button";

describe("Button", () => {
  const defProps: IProps = { onPress: jest.fn() };
  const mkWrapper = (propOverrides?: Partial<IProps>) => {
    const props = { ...defProps, ...propOverrides };
    return shallow(<Button {...props} />);
  };

  it("triggers `onPress` on press", () => {
    const { onPress } = defProps;
    const wrapper = mkWrapper();
    expect(onPress).not.toHaveBeenCalled();

    wrapper.simulate("press");

    expect(onPress).toHaveBeenCalled();
  });

  it("renders properly", () => {
    const props = { ...defProps, children: ["Button label"] };
    const tree = reactTestRenderer.create(<Button {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
