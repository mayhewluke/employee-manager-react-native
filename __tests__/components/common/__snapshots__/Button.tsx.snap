// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#007aff",
      "borderRadius": 5,
      "borderWidth": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#007aff",
        "fontSize": 16,
        "fontWeight": "600",
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    Button label
  </Text>
</View>
`;
